apache/zeppelin
# [ZEPPELIN-2133] All interpreters sometimes throw random Connection refuse errors

### What is this PR for?
When shiro authentication is enabled, then at times on restarting an interpreter doesn't kill(s) the remote interpreter process. 

### What type of PR is it?
[Bug Fix]

### What is the Jira issue?
* [ZEPPELIN-2133](https://issues.apache.org/jira/browse/ZEPPELIN-2133)

### How should this be tested?
 - enable shiro authentication
 - run atleast 2/3 different interpreter, wait for paragraph to complete
 - not restart all of these interpreter
 - repeat step 2
 - from command line do (in this example I'm testing if livy was killed or not) `ps aux | grep interpreter | grep livy` and observe there would be 2x processes running.

### Screenshots (if appropriate)
Before 
![process-test-before](https://cloud.githubusercontent.com/assets/674497/23089994/1fb0064c-f5b9-11e6-80fd-84b08e4e8663.gif)

After
![process-test-after](https://cloud.githubusercontent.com/assets/674497/23090016/ac31b552-f5b9-11e6-8f11-8a66aef08192.gif)

I'm using watch command `watch "ps aux | grep livy | grep interpreter.sh | grep -v aux"`


### Questions:
* Does the licenses files need update? N/A
* Is there breaking changes for older versions? N/A
* Does this needs documentation? N/A
